git_machete/cmd.py

```diff
@@ -648,31 +649,131 @@ def annotate(b: str, words: List[str]) -> None:
+        create_pull_request, derive_current_user_login, set_milestone_of_pull_request
+
+    base: Optional[str] = up_branch.get(head)
+    if not base:
+        raise MacheteException(f'Branch {head} does not have a parent branch (it is a root), base branch for the PR cannot be established')
+
     org: str
     repo: str
-    if len(org_name_for_github_remote) == 1:
-        org, repo = list(org_name_for_github_remote.values())[0]
-    elif len(org_name_for_github_remote) > 1:
-        if 'origin' in org_name_for_github_remote:
-            org, repo = org_name_for_github_remote['origin']
-        else:
-            raise MacheteException(f'Multiple non-origin remotes correspond to GitHub in this repository: '
-                                   f'{", ".join(org_name_for_github_remote.keys())}, aborting')
+    _, (org, repo) = derive_remote_and_github_org_and_repo(cli_ctxt)
+    current_user: Optional[str] = derive_current_user_login()
+    debug(cli_ctxt, f'create_github_pr({head})', f'organization is {org}, repository is {repo}')
+    debug(cli_ctxt, f'create_github_pr({head})', 'current GitHub user is ' + (current_user or '<none>'))
+
+    def slurp_file_or_empty(path: str) -> str:
+        try:
+            return open(path).read()
+        except IOError:
+            return ''
+
+    title: str = popen_git(cli_ctxt, "log", "-1", "--format=%s").strip()
+    description_path = get_git_subpath(cli_ctxt, 'info', 'description')
+    description: str = slurp_file_or_empty(description_path)
+
+    ok_str = ' <green><b>OK</b></green>'
+    print(fmt(f'Creating a {"draft " if draft else ""}PR from `{head}` to `{base}`...'), end='', flush=True)
+    pr: GitHubPullRequest = create_pull_request(org, repo, head=head, base=base, title=title, description=description, draft=draft)
+    print(fmt(f'{ok_str}, see <b>{pr.html_url}</b>'))
+
+    milestone_path: str = get_git_subpath(cli_ctxt, 'info', 'milestone')
+    milestone: str = slurp_file_or_empty(milestone_path).strip()
+    if milestone:
+        print(fmt(f'Setting milestone of PR #{pr.number} to {milestone}...'), end='', flush=True)
+        set_milestone_of_pull_request(org, repo, pr.number, milestone=milestone)
+        print(fmt(ok_str))
+
+    if current_user:
+        print(fmt(f'Adding `{current_user}` as assignee to PR #{pr.number}...'), end='', flush=True)
+        add_assignees_to_pull_request(org, repo, pr.number, [current_user])
+        print(fmt(ok_str))
+
+    reviewers_path = get_git_subpath(cli_ctxt, 'info', 'reviewers')
+    reviewers: List[str] = non_empty_lines(slurp_file_or_empty(reviewers_path))
+    if reviewers:
+        print(fmt(f'Adding {", ".join(f"`{r}`" for r in reviewers)} as reviewer{"s" if len(reviewers) > 1 else ""} to PR #{pr.number}...'), end='', flush=True)
+        add_reviewers_to_pull_request(org, repo, pr.number, reviewers)
+        print(fmt(ok_str))
+
+    annotations[head] = f'PR #{pr.number}'
```

Comment:
It would actually be *extremely* useful if this annotation were the full PR url (eg. `https://github.com/VirtusLab/git-machete/pull/130`) rather than just the PR number.

This is then auto-converted into a clickable link is more terminals, which makes for a _super_ slick one-click-from-machete-to-PR experience.

Reply:
Good idea... the only caveat I can see is that the output of `status`/`traverse` may become pretty cluttered :/ I'd make the annotation format controllable by a command-line switch... either a flag (`--url`?) or even a separate subcommand (`anno-pr-nums` vs `anno-pr-urls`?)

Reply:
@rifel123 could you take a look? this can be probably extracted to a separate issue